ydb/docs/en/core/yql/reference/yql-core/udf/list/knn.md

```diff
@@ -0,0 +1,100 @@
+## Introduction
+
+[Nearest Neighbor search](https://en.wikipedia.org/wiki/Nearest_neighbor_search) (NN) is an optimization task that consists in finding the closest point (or set of points) in a given dataset to a given query point. Closeness can be defined in terms of distance or similarity metrics.
+A generalization of the NN problem is the k-NN problem, where it's required to find the k nearest points to the query point. This can be useful in various applications such as image classification, recommendation systems, and more.
+https://en.wikipedia.org/wiki/Nearest_neighbor_search
+
+The solution of the problem is divided into two major subclasses of methods: exact and approximate. In this document, we will talk about the exact brute force method.
+
+The method is based on calculating the distance from the query point to every other point in the database. This algorithm, also known as the naive approach, has a complexity of `O(dn)`, where `n` is the number of points in the dataset, and `d` is its dimension.
+
+The advantage of the method is that there is no need for additional data structures, such as specialized vector indexes.
+The disadvantage is the need for a complete data search. But this disadvantage is insignificant in cases where data has been pre-filtered, for example, by user ID.
+
+Example:
+
+```sql
+SELECT id, fact, embedding FROM Facts
```

Comment:
```suggestion
DECLARE $TargetEmbedding AS List<Float>;
SELECT id, fact, embedding FROM Facts
```

Reply:
Otherwise, the query won't be valid.

Reply:
eb58508fbc525ff955239f046a5c120537268724